pal03377/Athena-New
# Test server deployment

### Motivation and Context
 
 
We want to be able to easily test PRs in this repository. Also, we want to prepare the continuous deployment of Athena in production.

 ### Description
 
This PR introduces the possibility of deploying a PR to Athena test server 3. It builds on #39 and especially https://github.com/ls1intum/Pyris/pull/5

Similar to https://github.com/ls1intum/Artemis and https://github.com/ls1intum/Pyris, you can now add a `deploy:athena-test3` GitHub label to any PR to deploy its branch to our test server. A GitHub action will change the label to `lock:athena-test3` or `deployment-error` depending on whether some error occurs. If there is already another PR that locks the test server, the deployment is denied.

 ### Steps for Testing
 
- (if existing) Remove any `lock` labels from this PR
- Add an example commit to this PR. I suggest you change the title of the playground page or something similar.
- Wait until **all** GitHub actions related to docker images are completed. This might take a while. Especially the ThemisML image takes a long time. I'm currently investigating why that is.
- Add the `deploy:athena-test3` label to this PR
- Wait for the deployment action to deploy to the test server
- Check for the change on https://athenetest1-03.ase.cit.tum.de (playground e.g.: https://athenetest1-03.ase.cit.tum.de/playground)

Note: The ThemisML module is currently broken on the test server. This is because it assumes access to a GPU. I'm currently trying to solve this, but until then, it will not be testable.

 ### Screenshots
 
 
-/-
